### PR TITLE
torch.nn.functional.scaled_dot_product_attention implementation

### PR DESCRIPTION
**Pull request by richagadgil**
https://pytorch.org/docs/stable/generated/torch.nn.functional.scaled_dot_product_attention.html#torch-nn-functional-scaled-dot-product-attention

### CODE REVIEW

File: tests/fx/converters/test_attn.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    query = torch.randn(qkv_shape).cuda()
+    key = torch.randn(qkv_shape).cuda()
+    value = torch.randn(qkv_shape).cuda()
+
+    L, S = query.size(-2), key.size(-2)
+
+    lengths = [random.randint(1, S) for i in range(0, L)]
+
+    if attn_mask == True:
+        attn_mask = torch.ones(L, S, dtype=torch.bool).cuda()
```
Comment (shivadbhavsar) [inline tests/fx/converters/test_attn.py:26]: add a test where the attn_mask isnt bool (case where you just add the values to the attn_bias)

File: py/torch_migraphx/fx/converters/acc_ops_converters.py [hunk 2]
```diff
@@ -2313,9 +2313,84 @@ def acc_ops_bitwise_and(mgx_module, node, _args, kwargs):
...
+@migraphx_converter(acc_ops.scaled_dot_product_attention)
+def acc_ops_scaled_dot_product_attention(mgx_module, node, args, kwargs):
+    query, key, value = kwargs['query'], kwargs['key'], kwargs['value']
+
+    # L, S = query.size(-2), key.size(-2)
+    L, S = query.shape().lens()[-2], key.shape().lens()[-2]
+
+    # scale_factor = 1 / math.sqrt(query.size(-1)) if scale is None else scale
+    import math
+    scale_factor = 1 / math.sqrt(query.shape().lens()[-1]) if kwargs.get("scale") is None else kwargs["scale"]
```
Comment (shivadbhavsar) [inline py/torch_migraphx/fx/converters/acc_ops_converters.py:2326]:
```suggestion
    scale_factor = 1 / torch.sqrt(query.shape().lens()[-1]) if kwargs.get("scale") is None else kwargs["scale"]
```

File: tests/fx/converters/test_attn.py (hunk 1 above)
Comment (richagadgil) [reply tests/fx/converters/test_attn.py:28]: Done

**Review by shivadbhavsar — CHANGES_REQUESTED** [review]
seems like pytest is blowing up because both test file names are the same, call one test_attn_fx.py and the other test_attn_dynamo.py like with all the other tests

File: tests/dynamo/converters/test_attn_dynamo.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import pytest
+import torch
+from dynamo_test_utils import FuncModule, FuncModuleFirstOut, convert_to_mgx, verify_outputs
+import random
+
+@pytest.mark.parametrize('op_alias', [
+    torch.ops.aten._scaled_dot_product_flash_attention.default,
+])
```
Comment (shivadbhavsar) [inline tests/dynamo/converters/test_attn_dynamo.py:8]:
```suggestion
@pytest.mark.parametrize('op_alias', [
    pytest.param(torch.ops.aten._scaled_dot_product_flash_attention.default, marks=pytest.mark.skip_min_torch_ver("2.3")),
])
```

**Review by shivadbhavsar — APPROVED** [review]

**Pull request merged**